src/plugins/telemetry/public/plugin.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Plugin, CoreStart, HttpStart } from '../../../core/public';
+
+import { TelemetrySender, TelemetryService, TelemetryNotifications } from './services';
+
+export interface TelemetryPluginStart {
+  telemetryService: TelemetryService;
+  telemetryNotifications: TelemetryNotifications;
+}
+
+export class TelemetryPlugin implements Plugin<void, TelemetryPluginStart> {
+  public setup() {}
+
+  public start({
+    injectedMetadata,
+    http,
+    notifications,
+    overlays,
+    application,
+  }: CoreStart): TelemetryPluginStart {
+    const isPluginEnabled = injectedMetadata.getInjectedVar('telemetryEnabled') as boolean;
```

Comment:
Let's call this something else because it's a bit misleading. The plugin is "enabled" at the Core level if it's running at all. Maybe `isTelemetrySendingEnabled` would be more accurate?